### ISSUE TITLE
`interface` vs `struct` in go lang

### ISSUE THREAD

**Issue by satyajitnayk** [satyajitnayk/load-balancer#1]
In Go (Golang), both interfaces and structs are essential constructs, but they serve different purposes and are used in different ways.

1. **Struct:**
   - A struct is a composite data type that groups together variables (fields) under a single name.
   - Fields within a struct can have different data types.
   - You can think of a struct as a way to create a custom data type with multiple attributes.
   - Structs are used for representing and organizing data.

   Example of a struct:

   ```go
   type Person struct {
       Name    string
       Age     int
       Address string
   }

   // Creating an instance of the struct
   person := Person{
       Name:    "John Doe",
       Age:     30,
       Address: "123 Main St",
   }
   ```

2. **Interface:**
   - An interface defines a set of method signatures but doesn't provide the method implementations.
   - Interfaces are used to define behavior rather than structure.
   - Any type that implements all the methods declared by an interface implicitly satisfies that interface.
   - Interfaces provide a way to achieve polymorphism in Go.

   Example of an interface:

   ```go
   type Speaker interface {
       Speak() string
   }

   // A type that implements the Speaker interface
   type Dog struct{}

   func (d Dog) Speak() string {
       return "Woof!"
   }
   ```

   In this example, `Dog` implicitly satisfies the `Speaker` interface because it has a `Speak` method with the correct signature.

In summary, structs are used for organizing and representing data, while interfaces are used for defining behavior and enabling polymorphism. It's common to use structs to model data structures and interfaces to define contracts that types can fulfill.
